pyoptsparse/pyOpt_utils.py

```diff
@@ -570,3 +574,39 @@ def _broadcast_to_array(name: str, value: ArrayType, n_values: int, allow_none:
     if not allow_none and any([i is None for i in value]):
         raise Error(f"The {name} argument cannot be 'None'.")
     return value
+
+
+def try_import_compiled_module_from_path(module_name: str, path: Optional[str] = None) -> Union[types.ModuleType, str]:
+    """
+    Attempt to import a module from a given path.
+
+    Parameters
+    ----------
+    module_name : str
+        The name of the module
+    path : Optional[str]
+        The path to import from. If None, the default ``sys.path`` is used.
+
+    Returns
+    -------
+    Union[types.ModuleType, str]
+        If importable, the imported module is returned.
+        If not importable, the error message is instead returned.
+    """
+    orig_path = sys.path
+    if path is not None:
+        path = os.path.abspath(os.path.expandvars(os.path.expanduser(path)))
+        sys.path = [path]
+    try:
+        module = importlib.import_module(module_name)
+    except ImportError as e:
+        if path is not None:
+            warnings.warn(
+                f"{module_name} module could not be imported from {path}.",
+                ImportWarning,
```

Comment:
Not related directly to this PR, but is `testflo` raising `ImportWarning` by default? I tested just this function locally and the string above (which can be pretty useful) is not printed out by default - see [here](https://docs.python.org/3/library/warnings.html#warning-categories)

Reply:
Hmm this is a good point. Maybe I will switch to `UserWarning`? I hate how a lot of warnings are not displayed by default.

Reply:
Yeah, that's the default but we can set it explicitly.